src/main/java/slimeknights/tconstruct/gadgets/item/slimesling/EnderSlimeSlingItem.java

```diff
@@ -29,13 +34,26 @@ public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity en
     float f = getForce(stack, timeLeft);
 
     Vector3d look = player.getLookVec();
+    double offX = look.x * f;
+    double offY = look.y * f + 1; // add extra to help with bad collisions
+    double offZ = look.z * f;
+
+    List<BlockPos> posToCheck = new ArrayList<>();
+    BlockPos posAttempt = new BlockPos(player.getPosX() + offX, player.getPosY() + offY, player.getPosZ() + offZ);
+    while (Math.abs(offX) > .5 || Math.abs(offY) > .5 || Math.abs(offZ) > .5) { // while not too close to player
+      posToCheck.add(posAttempt);
+      offX -= (Math.abs(offX) > .25 ? (offX >= 0 ? 1 : -1) * .25 : 0);
+      offY -= (Math.abs(offY) > .25 ? (offY >= 0 ? 1 : -1) * .25 : 0);
+      offZ -= (Math.abs(offZ) > .25 ? (offZ >= 0 ? 1 : -1) * .25 : 0);
+      posAttempt = new BlockPos(player.getPosX() + offX, player.getPosY() + offY, player.getPosZ() + offZ);
+    }
+
+    // filter to only teleportable blocks
+    posToCheck = posToCheck.stream().distinct().filter(block -> !worldIn.getBlockState(block).isSuffocating(worldIn, block)).collect(Collectors.toList());
```

Comment:
Switch to findFirst instead of collect. No need to build a whole list when we only want the first element.